test/hipcub/test_hipcub_block_exchange.cpp

```diff
@@ -331,18 +331,20 @@ TYPED_TEST(HipcubBlockExchangeTests, BlockedToWarpStriped)
     constexpr size_t block_size = TestFixture::params::block_size;
     constexpr size_t items_per_thread = TestFixture::params::items_per_thread;
     constexpr size_t items_per_block = block_size * items_per_thread;
+    const unsigned int current_device_warp_size = rocprim::host_warp_size();
     // Given block size not supported
     bool is_block_size_unsupported = block_size > test_utils::get_max_block_size();
 #ifdef HIPCUB_CUB_API
     // CUB does not support exchanges to/from warp-striped arrangements
     // for incomplete blocks (not divisible by warp size)
     // Workaround for nvcc warning: "dynamic initialization in unreachable code"
     // (not a simple if with compile-time expression)
-    is_block_size_unsupported |= block_size % HIPCUB_WARP_THREADS != 0;
+    is_block_size_unsupported |= block_size % current_device_warp_size != 0;
 #endif
     if(is_block_size_unsupported)
     {
-        return;
+        printf("Unsupported test block size: %zu.     Skipping test\n", block_size);
+        GTEST_SKIP();
```

Comment:
Is it working in the kernel code too? Should not we put the GTEST_SKIP before the kernel call to host side?

Reply:
I am blind. It's not in the kernel.